ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx

```diff
@@ -0,0 +1,529 @@
+/* eslint-disable no-void */
+import React, { useState, ReactElement } from 'react';
+import {
+    Card,
+    CardBody,
+    CardTitle,
+    Checkbox,
+    Form,
+    PageSection,
+    TextArea,
+    TextInput,
+    ToggleGroup,
+    ToggleGroupItem,
+} from '@patternfly/react-core';
+import * as yup from 'yup';
+
+import { NotifierIntegrationBase } from 'services/NotifierIntegrationsService';
+
+import usePageState from 'Containers/Integrations/hooks/usePageState';
+import FormMessage from 'Components/PatternFly/FormMessage';
+import FormCancelButton from 'Components/PatternFly/FormCancelButton';
+import FormTestButton from 'Components/PatternFly/FormTestButton';
+import FormSaveButton from 'Components/PatternFly/FormSaveButton';
+import useIntegrationForm from '../useIntegrationForm';
+import { IntegrationFormProps } from '../integrationFormTypes';
+
+import IntegrationFormActions from '../IntegrationFormActions';
+import FormLabelGroup from '../FormLabelGroup';
+
+export type MicrosoftSentinel = {
+    microsoftSentinel: {
+        logIngestionEndpoint: string;
+        directoryTenantId: string;
+        applicationClientId: string;
+        clientCertAuthConfig: {
+            clientCert: string;
+            privateKey: string;
+        };
+        secret: string;
+        alertDcrConfig: {
+            dataCollectionRuleId: string;
+            streamName: string;
+            enabled: boolean;
+        };
+        auditLogDcrConfig: {
+            dataCollectionRuleId: string;
+            streamName: string;
+            enabled: boolean;
+        };
+    };
+    type: 'microsoftSentinel';
+} & NotifierIntegrationBase;
+
+export type MicrosoftSentinelFormValues = {
+    notifier: MicrosoftSentinel;
+    updatePassword: boolean;
+};
+
+export const validationSchema = yup.object().shape({
+    notifier: yup.object().shape({
+        name: yup.string().trim().required('Email integration name is required'),
+        microsoftSentinel: yup.object().shape({
+            logIngestionEndpoint: yup
+                .string()
+                .trim()
+                .required('A log ingestion endpoint is required'),
+            directoryTenantId: yup.string().trim().required('A directory tenant ID is required'),
+            applicationClientId: yup
+                .string()
+                .trim()
+                .required('A application client ID is required'),
+            secret: yup.string().trim(),
+            clientCertAuthConfig: yup.object().shape({
+                clientCert: yup.string().trim(),
+                privateKey: yup.string().trim(),
```

Comment:
Although not a blocker to merge, a possible pattern to avoid so people do not imitate in the future:
* One side of coin: Find in Files 12 results in 9 files
* Other side of coin: without another method like `required()` does `trim()` mean anything?

Reply:
Regarding trim without any other method, I had originally put this end as a placeholder, before writing a custom validator function, but after Simon and I discussed, we decided to keep it simple, rather than possibly over validating on initial release of the integration.